From git@z Thu Jan  1 00:00:00 1970
Subject: [PATCH 1/2] platform/x86/intel/pmc: Mark arl_d3_fixup() and
 arl_resume() as static
From: Nathan Chancellor <nathan@kernel.org>
Date: Fri, 22 Dec 2023 11:27:01 -0700
Message-Id: <20231222-intel-pmc-missing-prototypes-v1-1-3f0d47377d4c@kernel.org>
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 7bit

With -Wmissing-prototypes, there are two instances flagged with no
prototypes:

  drivers/platform/x86/intel/pmc/arl.c:680:6: error: no previous prototype for 'arl_d3_fixup' [-Werror=missing-prototypes]
    680 | void arl_d3_fixup(void)
        |      ^~~~~~~~~~~~
  drivers/platform/x86/intel/pmc/arl.c:685:5: error: no previous prototype for 'arl_resume' [-Werror=missing-prototypes]
    685 | int arl_resume(struct pmc_dev *pmcdev)
        |     ^~~~~~~~~~
  cc1: all warnings being treated as errors

These functions are not used outside of this translation unit, so mark
them as static to fix the warning.

Fixes: f34dcf397286 ("platform/x86/intel/pmc: Add Arrow Lake S support to intel_pmc_core driver")
Reported-by: Stephen Rothwell <sfr@canb.auug.org.au>
Closes: https://lore.kernel.org/all/20231222135412.6bd796cc@canb.auug.org.au/
Link: https://lore.kernel.org/r/20231222-intel-pmc-missing-prototypes-v1-1-3f0d47377d4c@kernel.org
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 drivers/platform/x86/intel/pmc/arl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/platform/x86/intel/pmc/arl.c b/drivers/platform/x86/intel/pmc/arl.c
index c20506a9f4a6..9d873e02c412 100644
--- a/drivers/platform/x86/intel/pmc/arl.c
+++ b/drivers/platform/x86/intel/pmc/arl.c
@@ -677,12 +677,12 @@ static struct pmc_info arl_pmc_info_list[] = {
  * Set power state of select devices that do not have drivers to D3
  * so that they do not block Package C entry.
  */
-void arl_d3_fixup(void)
+static void arl_d3_fixup(void)
 {
 	pmc_core_set_device_d3(ARL_NPU_PCI_DEV);
 }
 
-int arl_resume(struct pmc_dev *pmcdev)
+static int arl_resume(struct pmc_dev *pmcdev)
 {
 	arl_d3_fixup();
 	return pmc_core_resume_common(pmcdev);
-- 
2.43.0

